### PR TITLE
automatic latest builds/trunk builds for testing new features

### PR DESCRIPTION
**Pull request by AssassinTee**
Added deployment to appveyor, circleci and travicci to enable automatic trunk builds.

os:
- win64
- win32
- unix
- osx (mac)

It adds 4 releases (for each os), which are automatically updated and overwritten by changes in the master branch.

All of this files need changes, because I can't use the OAuth tokens from your github-repository and can't interact with your TravisCi, CircleCI or appveyor account.

The trunk-builds only include client and server binaries (currently).

The test builds below might all fail.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -39,3 +38,48 @@ jobs:
...
+            sudo apt-get update -y
+            sudo apt-get install cmake libfreetype6-dev libsdl2-dev -y
+            git submodule update --init
+            go get -u github.com/tcnksm/ghr
+      - run:
+          name: Build bam
+          command: |
+            git clone https://github.com/matricks/bam.git ~/bam
+            cd ~/bam/
+            git reset --hard f012dd9a3e38295b8a45af5a101d29573381f169
```
Comment (Dune-jr) [inline .circleci/config.yml]: I think it is wisest to pull https://github.com/teeworlds/bam

Comment (AssassinTee) [reply .circleci/config.yml]: I copied that code from you, take a look at the building with bam part!

Comment (Dune-jr) [reply .circleci/config.yml]: I did not write that. It doesn't matter too much anyway :)

**Comment by oy** [comment]
What do we have to do here?

**Comment by oy** [comment]
@heinrich5991 should probably take a look at this, he set up the automatic build checks.

**Comment by heinrich5991** [comment]
Hmmm. I set it up so far and it'd probably work. However, it requires me to create a personal access token which would have full access to at least the Teeworlds repository. This means that in case we forgot some corner case, someone with a pull request can take over the Teeworlds repository; this is not a good solution. I guess I could try to set up some upload server (other than GitHub releases) that doesn't have this issue.

**Comment by oy** [comment]
Yeah, that would be good 👍

**Comment by oy** [comment]
Fixed by #2369

**Pull request closed without merging**